messaging/src/main/java/org/axonframework/eventhandling/scheduling/dbscheduler/DbSchedulerEventScheduler.java

```diff
@@ -0,0 +1,284 @@
+     * <p>
+     * The {@link Scheduler}, {@link Serializer} and {@link EventBus} are <b>hard requirements</b> and as such should be
+     * provided.
+     *
+     * @return a Builder to be able to create a {@link DbSchedulerEventScheduler}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public ScheduleToken schedule(Instant triggerDateTime, Object event) {
+        DbSchedulerScheduleToken taskInstanceId = new DbSchedulerScheduleToken(UUID.randomUUID().toString());
+        try {
+            DbSchedulerEventData data;
+            if (event instanceof EventMessage) {
+                data = detailsFromEvent((EventMessage<?>) event);
+            } else {
+                data = detailsFromObject(event);
+            }
+            TaskInstance<?> taskInstance = taskDescriptor.instance(taskInstanceId.getId(), data);
+            scheduler.schedule(taskInstance, triggerDateTime);
+        } catch (Exception e) {
+            throw new SchedulingException("An error occurred while scheduling an event.", e);
+        }
+        return taskInstanceId;
+    }
+
+    @SuppressWarnings("raw")
+    public static Task<DbSchedulerEventData> task() {
+        return new Tasks.OneTimeTaskBuilder<>(TASK_NAME, DbSchedulerEventData.class)
+                .execute((ti, context) -> {
+                    DbSchedulerEventScheduler eventScheduler = eventSchedulerReference.get();
+                    if (isNull(eventScheduler)) {
+                        throw new EventSchedulerNotSetException();
+                    }
+                    EventMessage<?> eventMessage = eventScheduler.fromDbSchedulerEventData(ti.getData());
+                    eventScheduler.publishEventMessage(eventMessage);
+                });
+    }
+
+    private DbSchedulerEventData detailsFromObject(Object event) {
+        SerializedObject<String> serialized = serializer.serialize(event, String.class);
+        String serializedPayload = serialized.getData();
+        String payloadClass = serialized.getType().getName();
+        String revision = serialized.getType().getRevision();
+        return new DbSchedulerEventData(serializedPayload, payloadClass, revision, null);
+    }
+
+    private DbSchedulerEventData detailsFromEvent(EventMessage<?> eventMessage) {
+        SerializedObject<String> serialized = serializer.serialize(eventMessage.getPayload(), String.class);
+        String serializedPayload = serialized.getData();
+        String payloadClass = serialized.getType().getName();
+        String revision = serialized.getType().getRevision();
+        String serializedMetadata = serializer.serialize(eventMessage.getMetaData(), String.class).getData();
+        return new DbSchedulerEventData(serializedPayload, payloadClass, revision, serializedMetadata);
+    }
+
+    private EventMessage<?> fromDbSchedulerEventData(DbSchedulerEventData data) {
```

Comment:
It seems it's not possible to publish domain events, is that right? I don't see persistence of aggregate identifier, type, and such

Reply:
That's correct, that's a limitation from the EventScheduler, from the interface: "In the latter case, the instance provided is the donor for the payload and Meta Data" Even the Axon server implementation only keeps the payload, type and revision.